LightFramework/src/rendering/scenerenderer.cpp

```diff
@@ -124,31 +164,49 @@ namespace Light {
 
 		m_framebuffer->clearAttachment(1, 0);
 
-		Light::Renderer::beginScene(camera, camera.getViewMatrix());
+		Light::Renderer::beginScene(camera, camera.getViewMatrix(), camera.calculatePosition());
 
-		std::vector<PointLight> lights;
+		std::vector<PointLight> pointLights;
+		std::vector<SpotLight> spotLights;
+		std::vector<DirectionalLight> directionalLights;
 		{
 			auto view = scene->m_registry.view<LightComponent, TransformComponent>();
-			for(auto& entity: view)
+
+			for (auto &entity : view)
 			{
-				auto[light, transform] = view.get(entity);
-				lights.push_back({transform.position, light.m_lightColor});
+
+				auto [light, transform] = view.get(entity);
+				switch ((int)light.m_lightType)
+				{
+				case 0:
+					directionalLights.push_back({glm::normalize(transform.getTransform() * glm::vec4(1.0, 1.0, 1.0, 0.0)), light.m_lightColor});
```

Comment:
If you are pushing back a newly constructed object, consider using `emplace_back` instead. Same for lines 185 and 188.

Reply:
The light objects don't really have a constructor before, so I didn't make changes here